Frontend/src/Patient/NewPatientModal.jsx

```diff
@@ -13,28 +15,25 @@ const NewPatientModal = ({ onClose, onCreate}) => {
   const [prescriptions, setPrescriptions] = useState([{ name: '', dose: '', instructions: '', date: '' }]);
   const [conditions, setConditions] = useState([{ name: '', date: '' }]);
   const [searchTerm, setSearchTerm] = useState('');
-  const [patients, setPatients] = useState([]);
-
+  let patientsData;
   useEffect(() => {
     const userId = user.id;
-    const storedUserData = localStorage.getItem('userData');
     async function fetchData() {
       try {
         const response = await fetch(`http://localhost:3000/users/${userId}/patients`, {
           method: 'GET',
         });
         if (!response.ok) {
           console.error('Failed to fetch patients:', response);
+          setPatients([]);
         } else {
-          const patientsData = await response.json();
+          patientsData = await response.json();
           setPatients(patientsData);
-          if (storedUserData) {
-            const userData = JSON.parse(storedUserData);
-            setUser({ ...userData, patients: patientsData });
-          }
+          user.patients = patientsData;
         }
       } catch (error) {
         console.error('Error fetching patients:', error);
```

Comment:
Instead of logging your errors to the console, you should handle the error gracefully in the UI. Consider having a modal or toast with the error message.